From 79d9f21cdc1429a197b6c7fb09bbdb69e6e03c66 Mon Sep 17 00:00:00 2001
From: Michael Bestas <mkbestas@lineageos.org>
Date: Fri, 25 May 2018 21:33:46 +0300
Subject: [PATCH 2/3] power: Remove unused camera_hint_ref_count

* These were part of the commit that added mutex locks
  for camera hints. We have now removed the locks so
  this is no longer necessary

Change-Id: I38b91ce4e8fc811b8d9a04e6ec20521adde1b264
---
 power-660.c  | 22 +++++++---------------
 power-8953.c | 23 +++++++----------------
 2 files changed, 14 insertions(+), 31 deletions(-)

diff --git a/power-660.c b/power-660.c
index 76ee891..d6a3003 100644
--- a/power-660.c
+++ b/power-660.c
@@ -52,7 +52,6 @@
 
 static int video_encode_hint_sent;
 
-static int camera_hint_ref_count;
 static void process_video_encode_hint(void *metadata);
 
 /**
@@ -239,23 +238,16 @@ static void process_video_encode_hint(void *metadata)
                 memcpy(resource_values, res, MIN_VAL(sizeof(resource_values), sizeof(res)));
                 num_resources = ARRAY_SIZE(res);
             }
-            camera_hint_ref_count++;
-            if (camera_hint_ref_count == 1) {
-                if (!video_encode_hint_sent) {
-                    perform_hint_action(video_encode_metadata.hint_id,
-                    resource_values, num_resources);
-                    video_encode_hint_sent = 1;
-                }
-           }
+            if (!video_encode_hint_sent) {
+                perform_hint_action(video_encode_metadata.hint_id,
+                        resource_values, num_resources);
+                video_encode_hint_sent = 1;
+            }
         }
     } else if (video_encode_metadata.state == 0) {
         if (is_interactive_governor(governor)) {
-            camera_hint_ref_count--;
-            if (!camera_hint_ref_count) {
-                undo_hint_action(video_encode_metadata.hint_id);
-                video_encode_hint_sent = 0;
-            }
-            return ;
+            undo_hint_action(video_encode_metadata.hint_id);
+            video_encode_hint_sent = 0;
         }
     }
     return;
diff --git a/power-8953.c b/power-8953.c
index cacf98b..bf9b777 100644
--- a/power-8953.c
+++ b/power-8953.c
@@ -50,7 +50,6 @@
 
 static int video_encode_hint_sent;
 
-static int camera_hint_ref_count;
 static void process_video_encode_hint(void *metadata);
 
 int  power_hint_override(power_hint_t hint, void *data)
@@ -152,24 +151,16 @@ static void process_video_encode_hint(void *metadata)
                                      0x41434000, 0x1,
                                      0x41424000, 0x28,
                                      };
-            camera_hint_ref_count++;
-            if (camera_hint_ref_count == 1) {
-                if (!video_encode_hint_sent) {
-                    perform_hint_action(video_encode_metadata.hint_id,
-                    resource_values,
-                    ARRAY_SIZE(resource_values));
-                    video_encode_hint_sent = 1;
-                }
-           }
+            if (!video_encode_hint_sent) {
+                perform_hint_action(video_encode_metadata.hint_id,
+                        resource_values, ARRAY_SIZE(resource_values));
+                video_encode_hint_sent = 1;
+            }
         }
     } else if (video_encode_metadata.state == 0) {
         if (is_interactive_governor(governor)) {
-            camera_hint_ref_count--;
-            if (!camera_hint_ref_count) {
-                undo_hint_action(video_encode_metadata.hint_id);
-                video_encode_hint_sent = 0;
-            }
-            return ;
+            undo_hint_action(video_encode_metadata.hint_id);
+            video_encode_hint_sent = 0;
         }
     }
     return;
-- 
2.17.0

